src/hooks/useReportIDs.tsx

```diff
@@ -0,0 +1,135 @@
+import React, {createContext, useCallback, useContext, useMemo} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxCollection, OnyxEntry} from 'react-native-onyx';
+import {getCurrentUserAccountID} from '@libs/actions/Report';
+import {getPolicyMembersByIdWithoutCurrentUser} from '@libs/PolicyUtils';
+import SidebarUtils from '@libs/SidebarUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Beta, Policy, PolicyMembers, PriorityMode, Report, ReportActions, TransactionViolation} from '@src/types/onyx';
+import useActiveWorkspace from './useActiveWorkspace';
+import useCurrentReportID from './useCurrentReportID';
+
+type OnyxProps = {
+    chatReports: OnyxCollection<Report>;
+    betas: OnyxEntry<Beta[]>;
+    policies: OnyxCollection<Policy>;
+    allReportActions: OnyxCollection<ReportActions>;
+    transactionViolations: OnyxCollection<TransactionViolation[]>;
+    policyMembers: OnyxCollection<PolicyMembers>;
+    priorityMode: OnyxEntry<PriorityMode>;
+};
+
+type WithReportIDsContextProviderProps = OnyxProps & {
+    children: React.ReactNode;
+    currentReportIDForTests?: string;
+};
+
+type ReportIDsContextValue = {
+    orderedReportIDs: string[];
+};
+
+const ReportIDsContext = createContext<ReportIDsContextValue>({
+    orderedReportIDs: [],
+});
+
+function WithReportIDsContextProvider({
+    children,
+    chatReports,
+    betas,
+    policies,
+    allReportActions,
+    transactionViolations,
+    policyMembers,
+    priorityMode,
+    /**
+     * Only required to make unit tests work, since we
+     * explicitly pass the currentReportID in LHNTestUtils
+     * to SidebarLinksData, so this context doesn't have an
+     * access to currentReportID in that case.
+     *
+     * This is a workaround to have currentReportID available in testing environment.
+     */
+    currentReportIDForTests,
+}: WithReportIDsContextProviderProps) {
+    const currentReportIDValue = useCurrentReportID();
+    const derivedCurrentReportID = currentReportIDForTests ?? currentReportIDValue?.currentReportID;
```

Comment:
```suggestion
    const currentReportIDValue = useCurrentReportID() ?? currentReportIDValue?.currentReportID;
```

Reply:
we can get rid of `derivedCurrentReportID`.

Reply:
I guess you mean:

```ts
const currentReportID = useCurrentReportID()?.currentReportID ?? currentReportIDForTests;
```

But we can't do it, since in the tests there's sometimes `reportID=0` and with above check we always have an empty string because `??` checks for `undefined` or `null`.

We can refactor this to use logical OR `||`:

```ts
const currentReportID = useCurrentReportID()?.currentReportID || currentReportIDForTests;
```

This works fine but the only issue is es-lint rule `prefer-nullish-coalescing`. We can silence this error if we want to use this approach.

Reply:
Answered [here](https://github.com/Expensify/App/pull/36420#discussion_r1529853929). The reason is that we have `0` as a value is because we are passing `currentReportID` as a prop rather through the `CurrentReportIDContextProvider`.

Reply:
ah I see your linked explanation makes sense thank you 😄